lib/common/xml.c

```diff
@@ -1682,6 +1684,9 @@ log_data_element(int log_level, const char *file, const char *function,
 
     char *prefix_m = NULL;
 
+    // @COMPAT: depth should become unsigned at a compatibility break
+    CRM_ASSERT(depth >= 0);
```

Comment:
This is effectively the same thing as changing the interface to `unsigned depth`, which I was trying to avoid for this public API function, so I wonder if I can do this either. On the one hand, if someone is passing a negative depth, that's a bug in their code. On the other, I think it would have "worked" before, in some form or fashion.

Reply:
The compatibility break is more about the function signature. Significant behavioral changes can be considered compatibility breaks as well, but I don't think this rises to that level, it's just improved error-checking.